mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ClientState.scala

```diff
@@ -50,7 +50,7 @@ import scala.util.{Either, Failure, Success}
             settings: MqttSessionSettings)(implicit mat: Materializer): Behavior[Event] =
     disconnected(
       Disconnected(
-        Vector.empty,
+        StashBuffer(Int.MaxValue),
```

Comment:
In Akka 2.6 milestones, the StashBuffer has been made `akka` private and is instead exposed as a behaviour with stash capabilities:

https://github.com/akka/akka/pull/27200

Can the stash functionality change be reverted here, as there is currently no way to cross-build StashBuffer usage across Akka 2.5 and 2.6.

Reply:
Clearly, StashBuffer is useful. I believe it was also recommended to us. We’ve put a ton of effort into this PR already, and tested it in production. Can it not be made public once more in Akka 2.6?

cc @patriknw

Reply:
@2m @patriknw I just had a chat with @longshorej about this. His response:

> I think there's more to it to be completely correct, StashBuffer provides methods to have behaviors receive all of the stashed events before processing the mailbox. The replacement would need to do that, similar to the code in QueueOfferState right now where the stashed messages are passed to interpretMessage/interpretSignal.

I think that the simplest path forward is to make `StashBuffer` public once more.

Reply:
The visibility of `StashBuffer` is not the main problem here. Alpakka modules live in the `akka` package namespace. Therefore we can access the internals of Akka.

The problem here is not having a single factory method for `StashBuffer` across Akka 2.5 and 2.6. This prevents this module to be easily cross-compilable across different Akka versions.

I see that in Akka 2.6 public stash buffer API has been refactored from a standalone data structure to only being accessible via `Behaviors.withStash`.

And the internal API has been modified to take ActorContext which it did not in Akka 2.5.

@patriknw Can we get a constructor of `StashBuffer` that takes `ActorContext` in Akka 2.5 as well?